CMakeLists.txt

```diff
@@ -9,6 +9,26 @@ set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin)
 add_compile_options(-Wall -Wextra -Wno-unused-parameter -Wno-unused-result -Werror)
 
+# Add Address Sanitiezed build type
+set(CMAKE_C_FLAGS_ASAN
+    "-g -ggdb -fsanitize=address -fno-omit-frame-pointer"
```

Comment:
I wouldn't add `-ggdb` to a build since it is not portable - obviously, it adds info that is gdb-only. Also, I'd base it on `CMAKE_C_FLAGS_DEBUG` and add just `-fsanitize=address`, and you don't need any of the below added lines.

Reply:
Updated